<launch>
  <!-- motor node -->
  <include file="$(find xycar_motor)/launch/xycar_motor_a2.launch" />

  <!-- usb_cam -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/videoCAM" />
    <param name="autoexposure" value="false"/>
    <param name="exposure" value="80"/>
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="camera_name" value="usb_cam" />
    <param name="camera_info_url" value="file://$(find final_jh)/calibration/usb_cam_2.yaml" />
  </node>

  <node name="xycar_lidar"	pkg="xycar_lidar"	type="xycar_lidar"	output="screen">
    <param name="serial_port"	type="string"	value="/dev/ttyRPL"/>
    <param name="serial_baudrate"	type="int"	value="115200"/>
    <param name="frame_id"		type="string"	value="laser"/>
    <param name="inverted"		type="bool"	value="false"/>
    <param name="angle_compensate"	type="bool"	value="true"/>
  </node>

  <!-- ar track-->
  <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
    <param name="marker_size" type="double" value="9.0" />
    <param name="max_new_marker_error" type="double" value="0.05" />
    <param name="max_track_error" type="double" value="0.05" />
    <param name="output_frame" type="string" value="usb_cam" />
    <remap from="camera_image" to="/usb_cam/image_raw" />
    <remap from="camera_info" to="/usb_cam/camera_info" />
  </node>

  <!-- auto drive -->
  <node name="auto_drive" pkg="final_jh" type="autodrive.py" output="screen"/>
  <node name="hough_drive" pkg="final_jh" type="hough_drive_a2.py" output="screen"/>
  <node name="ar_drive" pkg="final_jh" type="ar_info_print.py" output="screen"/>
  <node name="lidar_driver"	pkg="final_jh"	type="lidar_gostop.py"	output="screen"/> -->

</launch>
